daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/BitFunctions.tdml

```diff
@@ -0,0 +1,451 @@
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="rightShiftLongError02" root="rightLong" model="BitFunctions">
+              <tdml:document>2,64</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Right shift not supported for shift greater or equal to 64</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="leftShiftShortError01" root="leftShort" model="BitFunctions">
+              <tdml:document>2.2,2</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Unable to parse xs:short from text: 2.2</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+  <tdml:parserTestCase name="leftShiftShortError02" root="leftShort" model="BitFunctions">
+              <tdml:document>2,16</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Left shift not supported for shift greater or equal to 16</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="rightShiftShortError01" root="rightShort" model="BitFunctions">
+              <tdml:document>2.2,2</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Unable to parse xs:short from text: 2.2</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="rightShiftShortError02" root="rightShort" model="BitFunctions">
+              <tdml:document>2,16</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Right shift not supported for shift greater or equal to 16</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+
+  <tdml:parserTestCase name="leftShiftByteError01" root="leftByte" model="BitFunctions">
+              <tdml:document>2.2,2</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Unable to parse xs:byte from text: 2.2</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+  <tdml:parserTestCase name="leftShiftByteError02" root="leftByte" model="BitFunctions">
+              <tdml:document>2,8</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Left shift not supported for shift greater or equal to 8</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="rightShiftByteError01" root="rightByte" model="BitFunctions">
+              <tdml:document>2.2,2</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Unable to parse xs:byte from text: 2.2</tdml:error>
+                </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="rightShiftByteError02" root="rightByte" model="BitFunctions">
+              <tdml:document>2,8</tdml:document>
+                <tdml:errors>
+                  <tdml:error>Right shift not supported for shift greater or equal to 8</tdml:error>
```

Comment:
Negative tests for shift values too large look good. Might also consider adding some zero or negative shift values.

Would also like to see tests for unsigned types, especially around the cornder cases (e.g. shifting left a max value)--really need to ensure we are handling those correctly